tests/src/tracing/tracevalidation/inducedgc/InducedGC.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.IO;
+using Tracing.Tests.Common;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+
+namespace Tracing.Tests
+{
+    public static class TraceValidationInducedGC
+    {
+        private static int InducedGCIterations = 10;
+
+        public static int Main(string[] args)
+        {
+            bool pass = true;
+            bool keepOutput = false;
+
+            // Use the first arg as an output filename if there is one.
+            string outputFilename = null;
+            if (args.Length >= 1)
+            {
+                outputFilename = args[0];
+                keepOutput = true;
+            }
+            else
+            {
+                outputFilename = System.IO.Path.GetTempPath() + Guid.NewGuid().ToString() + ".netperf";
+            }
+
+            try
+            {
+                Console.WriteLine("\tStart: Enable tracing.");
+                TraceControl.EnableDefault(outputFilename);
+                Console.WriteLine("\tEnd: Enable tracing.\n");
+
+                Console.WriteLine("\tStart: Generate some events.");
+                for(int i=0; i<InducedGCIterations; i++)
+                {
+                    GC.Collect();
+                }
+                Console.WriteLine("\tEnd: Generate some events.\n");
+
+                Console.WriteLine("\tStart: Disable tracing.");
+                TraceControl.Disable();
+                Console.WriteLine("\tEnd: Disable tracing.\n");
+
+                Console.WriteLine("\tStart: Process the trace file.");
+                int matchingEventCount = 0;
+                int nonMatchingEventCount = 0;
+                using (var trace = TraceEventDispatcher.GetDispatcherFromFileName(outputFilename))
+                {
+                    string gcReasonInduced = GCReason.Induced.ToString();
+                    string providerName = "Microsoft-Windows-DotNETRuntime";
+                    string gcTriggeredEventName = "GC/Triggered";
+
+                    trace.Clr.GCTriggered += delegate(GCTriggeredTraceData data)
+                    {
+                        if(gcReasonInduced.Equals(data.Reason.ToString()) &&
+                           providerName.Equals(data.ProviderName) &&
+                           gcTriggeredEventName.Equals(data.EventName))
+                        {
+                            matchingEventCount++;
+                        }
+                        else
+                        {
+                            nonMatchingEventCount++;
+                        }
+                    };
+
+                    trace.Process();
```

Comment:
to get more information for troubleshooting we could also attach to `eventSource.UnhandledEvents` handler. Is it possible to somehow write all unhandled events to a file and later on download it from the CI machine?

Reply:
This is interesting.  I need to look into what events fall into UnhandledEvents to know if this can test EventPipe functionality or if it's really a TraceEvent test that belongs in the PerfView repo.

I'm not sure if our current CI jobs will archive files that the test run produces, but I imagine they could be made to (or we could create our own job if needed).

Reply:
If you want to enable failed run diagnostics you can also get the whole file out so that you could look at it on your local machine. That way we could more carefully figure out corruption or the stream, dropped events, or any other cause of failure.

Reply:
Thanks @nategraf.  I will follow-up on this, likely for a subsequent PR.  It would be great to get files out for failed tests.